### ISSUE TITLE
Fatal ESLint error in fxa-payments-server/src/lib/stripe.ts when using global _dev/.eslint config

### ISSUE THREAD

**Issue by pdehaan** [mozilla/fxa#11732]
## Description

This is a bit weird, and possibly worthy of some deeper thought than I can manage&hellip;
In https://github.com/mozilla/fxa/pull/11729 PR I tweaked the `"lint"` package.json script to lint the code.

I'm seeing some ESLint errors when trying to lint the codebase. Although I only see it when I try and run the code from the root directory and not if I run <kbd>yarn lint</kbd> from within packages/fxa-payments-server. WHEW!

```sh
pwd
# /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server

yarn lint
# ✖ 214 problems (63 errors, 151 warnings)
#  52 errors and 14 warnings potentially fixable with the `--fix` option.
```

But when I was trying to lint the files from the root directory, I get different errors:

```sh
DEBUG=eslint:* npx eslint packages/fxa-payments-server -c _dev/.eslintrc --ext=.{js,jsx,ts,tsx} 2> fxa-payments-server.eslint.log
```

And the last few dozen lines of fxa-payments-server.eslint.log look like this (the full log file is over 137K lines long):
```sh
2022-01-24T22:58:25.775Z eslint:linter An error occurred while traversing
2022-01-24T22:58:25.775Z eslint:linter Filename: /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/stripe.ts
2022-01-24T22:58:25.775Z eslint:linter Line: 5
2022-01-24T22:58:25.775Z eslint:linter Parser Options: {
  ecmaVersion: 11,
  ecmaFeatures: { globalReturn: false, jsx: true },
  sourceType: 'module',
  warnOnUnsupportedTypeScriptVersion: true
}
2022-01-24T22:58:25.776Z eslint:linter Parser Path: /Volumes/Dev/github/mozilla/fxa/node_modules/@typescript-eslint/parser/dist/index.js
2022-01-24T22:58:25.776Z eslint:linter Settings: { react: { version: 'detect' } }

Oops! Something went wrong! :(

ESLint: 7.32.0

AssertionError [ERR_ASSERTION]: Node must be provided when reporting error if location is not provided
    at assertValidNodeInfo (/Volumes/Dev/github/mozilla/fxa/node_modules/eslint/lib/linter/report-translator.js:98:9)
    at /Volumes/Dev/github/mozilla/fxa/node_modules/eslint/lib/linter/report-translator.js:332:9
    at Object.report (/Volumes/Dev/github/mozilla/fxa/node_modules/eslint/lib/linter/linter.js:926:41)
    at Program:exit (/Volumes/Dev/github/mozilla/fxa/node_modules/eslint/lib/rules/no-unused-vars.js:667:33)
    at /Volumes/Dev/github/mozilla/fxa/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/Volumes/Dev/github/mozilla/fxa/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/Volumes/Dev/github/mozilla/fxa/node_modules/eslint/lib/linter/node-event-generator.js:293:26)
    at NodeEventGenerator.applySelectors (/Volumes/Dev/github/mozilla/fxa/node_modules/eslint/lib/linter/node-event-generator.js:322:22)
    at NodeEventGenerator.leaveNode (/Volumes/Dev/github/mozilla/fxa/node_modules/eslint/lib/linter/node-event-generator.js:345:14)
```

There isn't a nested .eslintrc config file in the ./packages/fxa-payments-server/ folder, but there _is_ an `eslintConfig` property:

```js
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
```

I'm still trying to wrap my head around that error and why I only see it in one of the linting approaches. Not sure if it's because the fxa-payments-server repo is possibly ONLY extending from "react-app" and "react-app/jest" configs and not the base _dev/.eslintrc configs in this one specific package. Will need a bit more poking. I think I can probably try and use ESLint's `--print-config` flag to inspect a specific file within the fxa-payments-server package and see if it inherits any fxa/default settings or not. Or see if any other packages also use an `eslintConfig` key in the package.json and if that affects config inheritance at all.

Anyways, just thought I'd live-blog this while I saw it in case it is or isn't a known issue.

### Steps to reproduce

TLDR:

| COMMAND | RESULT | NOTES
|:----|:----|:----
| <kbd>cd packages/fxa-payments-server; yarn lint</kbd> | 214 problems (63 errors, 151 warnings) | Errors in .js, .ts, and .tsx files, so it is linting all the TypeScript.
| <kbd>yarn lint</kbd> (from project root) | 214 problems (63 errors, 151 warnings) | No surprise, it calls the package's `yarn lint` command.
| <kbd>npx eslint packages/fxa-payments-server --ext=.{js,jsx,ts,tsx} | 214 problems (63 errors, 151 warnings) | Same result, seems to be using the packages/fxa-payments-server/package.json `eslintConfig` config.
| <kbd>npx eslint packages/fxa-payments-server --ext=.{js,jsx,ts,tsx} -c _dev/.eslintrc</kbd> | AssertionError [ERR_ASSERTION]: Node must be provided when reporting error if location is not provided | ESLint doesn't work in this package w/ shared ESLint config.

### Expected result

I think we need to find a way where we can:
- lint the files via git commit-hooks (via Husky and lint-staged)
- lint the files globally via a root <kbd>yarn lint</kbd> and it calls each packages's <kbd>lint</kbd> script
- lint a specific packages's files using <kbd>yarn lint</kbd> from the package directory (instead of relying on git hooks only, which can contain bugs if nobody ever modifies those files; or somebody just bypasses hooks and force pushes).
- have all of the above methods use the same configuration so the git hooks isn't using one config file but other linting uses a different set of rules.

### Actual result

What happened.

### Environment

What browser and operating system are you using?

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-4484)

**Comment by pdehaan** [comment]
**DISCLAIMER:** I honestly have zero idea what's actually happening above. It doesn't seem to affect ALL files in that repo, or all .js, .ts, or all .tsx files, potentially something in that one _specific_ file. Here was my command and output when I found the issue (except I was linting "packages/", not the more specific "packages/fxa-payments-server"):

```sh
DEBUG=eslint:cli-engine npx eslint packages/fxa-payments-server -c _dev/.eslintrc --ext=.{js,jsx,ts,tsx}

  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/fxa-content-server-l10n/Gruntfile.js +0ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/fxa-content-server-l10n/grunttasks/l10n-lint.js +68ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/fxa-content-server-l10n/scripts/ci-creds.js +25ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/jest.config.js +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/pm2.config.js +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/bin/fxa-payments-server.js +13ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/config/index.js +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/jest.config.js +17ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/404.js +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/amplitude.js +8ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/amplitude.test.js +15ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/csp/blocking.js +30ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/csp/report-only.js +13ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/csp.js +2ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/csp.test.js +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/html-middleware.js +9ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/logging/log.js +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/logging/route-logging.js +2ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/no-robots.js +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/routes/index.js +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/routes/index.test.js +2ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/routes/legal-docs.js +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/routes/legal-docs.test.js +15ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/routes/navigation-timing.js +20ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/routes/navigation-timing.test.js +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/routes/post-metrics.js +9ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/routes/post-metrics.test.js +7ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/server.js +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/server.test.js +19ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/server/lib/version.js +8ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/App.test.tsx +6ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/App.tsx +53ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/AlertBar/index.stories.tsx +24ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/AlertBar/index.test.tsx +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/AlertBar/index.tsx +7ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/AppLayout/index.stories.tsx +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/AppLayout/index.test.tsx +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/AppLayout/index.tsx +12ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/CouponForm/index.stories.tsx +11ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/CouponForm/index.test.tsx +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/CouponForm/index.tsx +10ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/DialogMessage/index.stories.tsx +13ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/DialogMessage/index.test.tsx +11ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/DialogMessage/index.tsx +6ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/ErrorMessage/index.tsx +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/FetchErrorDialogMessage.tsx +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/Header/index.stories.tsx +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/Header/index.test.tsx +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/Header/index.tsx +9ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/LoadingOverlay/index.stories.tsx +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/LoadingOverlay/index.test.tsx +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/LoadingOverlay/index.tsx +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/LoadingSpinner/index.test.tsx +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/LoadingSpinner/index.tsx +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/NewUserEmailForm/index.stories.tsx +2ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx +15ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PayPalButton/index.stories.tsx +13ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PayPalButton/index.test.tsx +6ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PayPalButton/index.tsx +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentConfirmation/index.stories.tsx +13ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentConfirmation/index.test.tsx +7ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx +29ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentConsentCheckbox/index.stories.tsx +8ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentConsentCheckbox/index.test.tsx +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentConsentCheckbox/index.tsx +17ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentErrorView/index.stories.tsx +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentErrorView/index.test.tsx +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentErrorView/index.tsx +8ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentForm/index.stories.tsx +8ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentForm/index.test.tsx +11ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentForm/index.tsx +21ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentLegalBlurb/index.stories.tsx +18ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentLegalBlurb/index.tsx +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentMethodHeader/index.stories.tsx +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentMethodHeader/index.test.tsx +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentMethodHeader/index.tsx +8ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentProcessing/index.stories.tsx +6ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentProcessing/index.test.tsx +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentProcessing/index.tsx +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentProviderDetails/index.test.tsx +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PaymentProviderDetails/index.tsx +6ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PlanDetails/index.stories.tsx +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PlanDetails/index.test.tsx +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PlanDetails/index.tsx +26ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PlanErrorDialog/index.stories.tsx +10ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PlanErrorDialog/index.test.tsx +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/PlanErrorDialog/index.tsx +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/SubscriptionTitle/index.stories.tsx +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/SubscriptionTitle/index.test.tsx +2ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/SubscriptionTitle/index.tsx +7ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/TermsAndPrivacy/index.stories.tsx +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/TermsAndPrivacy/index.test.tsx +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/TermsAndPrivacy/index.tsx +8ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/Tooltip/index.stories.tsx +7ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/Tooltip/index.test.tsx +7ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/Tooltip/index.tsx +11ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/fields/index.test.tsx +7ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/components/fields/index.tsx +34ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/images/fpn.js +16ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/index.test.tsx +2ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/index.tsx +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/AppContext.test.tsx +7ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/AppContext.tsx +6ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/Coupon.ts +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/PaymentProvider.test.ts +2ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/PaymentProvider.ts +6ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/account.test.ts +8ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/account.ts +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/amplitude.test.ts +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/amplitude.ts +7ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/apiClient.test.ts +9ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/apiClient.ts +37ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/config.test.ts +14ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/config.ts +10ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/customer.test.ts +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/customer.ts +8ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/errors.test.tsx +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/errors.ts +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/flow-event.test.ts +5ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/flow-event.ts +7ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/format.test.ts +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/formats.ts +7ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/hooks.test.tsx +9ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/hooks.tsx +6ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/logger.js +7ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/logger.test.js +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/mock-data.tsx +9ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/newsletter.test.ts +7ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/newsletter.ts +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/plan.ts +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/react-test-renderer.d.ts +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/screen-info.test.ts +2ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/screen-info.ts +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/sentry.js +4ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/sentry.test.js +8ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/speed-trap.d.ts +8ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/stripe.test.ts +3ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/fxa/packages/fxa-payments-server/src/lib/stripe.ts +6ms

Oops! Something went wrong! :(

ESLint: 7.32.0

AssertionError [ERR_ASSERTION]: Node must be provided when reporting error if location is not provided
...
```

So not sure what specifically in packages/fxa-payments-server/src/lib/stripe.ts is triggering this. 🤷

**Comment by pdehaan** [comment]
Not sure the "eslintConfig" in package.json is relevant, looks like a few other packages have that as well:

```sh
git grep "\"eslintConfig\"" packages

packages/fxa-admin-panel/package.json:  "eslintConfig": {
packages/fxa-payments-server/package.json:  "eslintConfig": {
packages/fxa-settings/package.json:  "eslintConfig": {
```

**Comment by data-sync-user** [comment]
➤ Reino Muhl commented:

We didn’t add the Subscription Platform component to this ticket, since it seems the suggestions are more general for the entire FXA repo.

To fix the linting errors in fxa-payments-server specifically, we have this ticket. FXA-4403 ( https://mozilla-hub.atlassian.net/browse/FXA-4403 )

**Comment by cknowles-admin** [comment]
FxA is no longer synchronizing all issues between Jira and Github.  We are closing open issues and will selectively synchronize in the future.
